prow/plugins/skip/skip.go

```diff
@@ -86,6 +88,14 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 		log.Warn(resp)
 		return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
 	}
+	baseSHA, err := gc.GetRef(org, repo, "heads/"+pr.Base.Ref)
```

Comment:
So do we just think that the closures are not necessary for these since they're much less common?

Reply:
Yeah.

Going a step back, we could change the whole thing to instead have `GetPresubmits` in the `config` package take closures, that way we would keep all of that in one place in the config package.

The main drawback of that is that the callers basically must retain a pointer to whatever gets filled in the clojure and fill it themselves if its nil in order to make sure we don't use different SHAs for getting the Presubmits and the rest because of unlucky timing

Reply:
Yeah, that's kind of where I was going with my comment about somehow putting this into the client agent container. I think we could probably invent a structure that would hold the closures and their outputs so caller code could instantiate one of those then use it as a proxy for `GetPresubmits()` and/or `GetPullRequest()`, `GetRef()`. That just kicks the can down the road, meaning there's no authoritative way to ensure a plugin appropriately uses this new thing instead of calling `GetRef()` alone, so it's going to fall to review in any case. 

Actually, thinking about this once more -- does it make sense to make the `GetRef()` call at all here? If the ref changed, we're going to get another event into `hook` anyway. So for instance for the `/skip` command the only utility to reaching out for `GetRef()` in the handler is if the comment was posted first, then the PR was updated and we are handling the `/skip` out-of-order with the push event?